### DISCUSSION TITLE
Release candidates

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1067] [General]
Last release we ran an RC version for a while. The recent release of secp has some bugs [0] in it that we could have caught if we had done more thorough testing (build rust-bitcoin with the new version would have been enough). 

Should we consider doing an RC release for `rust-secp256k1` next time to save us getting so much egg on our faces? If we start crate smashing are we going to have to do RC for every crate, or is there a better way?

#### [0] 
- https://github.com/rust-bitcoin/rust-secp256k1/pull/466
- https://github.com/rust-bitcoin/rust-secp256k1/pull/460

**Comment by apoelstra** [comment]
I don't think it's worth going through the pain of a RC process just to deal with compilation errors that occur when you set compiler flags (not even feature flags) that disable all the crypto. I also don't think there's "egg on our faces" for an issue that only affects a couple specific downstream projects that we maintain, and which can be fixed with another point release.

**Reply by tcharding** [reply]
cool, I get a bit dramatic when mistakes are made that I feel responsible for :) Perhaps when the 'release X.Y.Z' PR goes up in any crate (below rust-miniscript) we (read: I) can throw up a branch that uses it in a crate above it. With a link from the tracking PR to the 'testing' PR we can see its green before proceeding?

**Reply by apoelstra** [reply]
That would be amazing if you're willing to do this manual work.

I suppose specifically with rust-secp and rust-bitcoin, we could put a CI job into rust-secp which checks this. Same with bitcoin_hashes. It may be a bit tricky to make sure this doesn't undermine our ability to make breaking/major-rev changes.

**Reply by tcharding** [reply]
Yes, I think its a PITA to have it in CI, I'm happy to do it manually, it helps makes sure I _fully_ understand all breaking changes that have gone in (e.g., the current `Scalar` from hash thing).
